verilog/analysis/extractors.cc

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/analysis/extractors.h"
+
+#include "verilog/analysis/verilog_analyzer.h"
+#include "verilog/CST/module.h"
+#include "verilog/CST/identifier.h"
+
+#include <string>
+
+namespace verilog {
+namespace analysis {
+
+// Find all modules and collect interface names
+absl::Status CollectInterfaceNames(absl::string_view content,
+                                   std::set<std::string>* if_names) {
+  VLOG(1) << __FUNCTION__;
+
+  const auto analyzer =
+      verilog::VerilogAnalyzer::AnalyzeAutomaticMode(content, NULL);
+  const auto lex_status = ABSL_DIE_IF_NULL(analyzer)->LexStatus();
+  const auto parse_status = analyzer->ParseStatus();
+
+  if (!lex_status.ok()) return lex_status;
+  if (!parse_status.ok()) return parse_status;
```

Comment:
FYI, it is possible to have a syntax error inside the module (could be false positive in the case of parser bugs), and still have a partial syntax tree to work with.  Up to you whether or not you want to allow this, but rejecting on syntax error is ok too.

Reply:
Added TODO to not always rejecting syntax error